### PR TITLE
Bump up dependency versions to latest majors

### PR DESCRIPTION
**Pull request by tuommaki**
In order to allow easier patches for the near future, bump up all
dependencies to latest major as we are also about to release a new major
version of azure-operator.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,28 +11,30 @@ require (
...
 	github.com/coreos/go-semver v0.3.0
 	github.com/giantswarm/apiextensions/v2 v2.6.2
-	github.com/giantswarm/apiextensions/v3 v3.4.0
+	github.com/giantswarm/apiextensions/v3 v3.7.0
 	github.com/giantswarm/appcatalog v0.2.7
 	github.com/giantswarm/apprclient/v2 v2.0.0
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/certs/v3 v3.1.0
-	github.com/giantswarm/e2e-harness/v2 v2.0.0
+	github.com/giantswarm/e2e-harness/v2 v2.0.1-0.20201105124938-04fb4b181f2b
```
Comment (tuommaki) [inline go.mod:19]: *TODO*: Need to release new major version once this is tested.

**Review by whites11 — APPROVED** [review]
sounds good when tested

**Review by nprokopic — APPROVED** [review]
LGTM when tested

**Comment by tuommaki** [comment]
Cluster creation works (including core apps & stuff). I consider this functional. Upgrade testing will follow up on separate branches. E2e is somehow funky, but I'm not stopping on this now for that.

**Pull request merged**